linera-sdk/src/lib.rs

```diff
@@ -216,51 +216,39 @@ pub trait Contract: WithContractAbi + ContractAbi + Send + Sized {
     ) -> Result<SessionCallResult<Self::Message, Self::Response, Self::SessionState>, Self::Error>;
 
     /// Calls another application.
-    // TODO(#488): Currently, the application state is persisted before the call and restored after the call in
-    // order to allow reentrant calls to use the most up-to-date state.
-    async fn call_application<A: ContractAbi + Send>(
+    fn call_application<A: ContractAbi + Send>(
         &mut self,
```

Comment:
Nice! We have now a few refactoring options to discuss for a next PR (e.g. merge the function with `crate::contract::system_api::call_application` and/or make it a method of `*Context`).